zerver/actions/default_streams.py

```diff
@@ -70,15 +71,15 @@ def do_add_default_stream(stream: Stream) -> None:
     stream_id = stream.id
     if not DefaultStream.objects.filter(realm_id=realm_id, stream_id=stream_id).exists():
         DefaultStream.objects.create(realm_id=realm_id, stream_id=stream_id)
-        notify_default_streams(stream.realm)
+        notify_default_streams(stream.realm, stream_id)
 
 
 @transaction.atomic(savepoint=False)
 def do_remove_default_stream(stream: Stream) -> None:
     realm_id = stream.realm_id
     stream_id = stream.id
     DefaultStream.objects.filter(realm_id=realm_id, stream_id=stream_id).delete()
-    notify_default_streams(stream.realm)
+    notify_default_streams(stream.realm, stream_id)
```

Comment:
Why this event is modified? We already pass all default streams.

Reply:
I see you have added this to update the stream settings UI. I think this can be completely handled in the UI only without any change in the event object. You can just check whether the stream opened in the stream settings UI is present in the default streams list already passed in this event, and then update the checkbox state accordingly.